corehq/apps/data_interfaces/tasks.py

```diff
@@ -110,27 +110,32 @@ def check_data_migration_in_progress(domain, last_migration_check_time):
 def run_case_update_rules_for_domain(domain, now=None):
     now = now or datetime.utcnow()
 
-    run_record = DomainCaseRuleRun.objects.create(
-        domain=domain,
-        started_on=datetime.utcnow(),
-        status=DomainCaseRuleRun.STATUS_RUNNING,
-    )
+    domain_rules = AutomaticUpdateRule.by_domain(domain, AutomaticUpdateRule.WORKFLOW_CASE_UPDATE)
+    all_rule_case_types = set(domain_rules.values_list('case_type', flat=True))
 
-    if should_use_sql_backend(domain):
-        for db in get_db_aliases_for_partitioned_query():
-            run_case_update_rules_for_domain_and_db.delay(domain, now, run_record.pk, db=db)
-    else:
-        # explicitly pass db=None so that the serial task decorator has access to db in the key generation
-        run_case_update_rules_for_domain_and_db.delay(domain, now, run_record.pk, db=None)
+    for case_type in all_rule_case_types:
+        run_record = DomainCaseRuleRun.objects.create(
+            domain=domain,
+            started_on=datetime.utcnow(),
+            status=DomainCaseRuleRun.STATUS_RUNNING,
+            case_type=case_type
+        )
+
+        if should_use_sql_backend(domain):
+            for db in get_db_aliases_for_partitioned_query():
+                run_case_update_rules_for_domain_and_db.delay(domain, now, run_record.pk, case_type, db=db)
+        else:
+            # explicitly pass db=None so that the serial task decorator has access to db in the key generation
+            run_case_update_rules_for_domain_and_db.delay(domain, now, run_record.pk, case_type, db=None)
 
 
 @serial_task(
-    '{domain}-{db}',
+    '{domain}-{case-type}-{db}',
```

Comment:
I think this could cause the same domain and case type to have two processes running at the same time. Also should it be `case_type`?

Reply:
yes, should be `case_type`

Do you think it's an issue that there would be multiple processes running at the same time?

Reply:
Oh sorry I was not clear at all. I think what I meant was:

1. queueing task is kicked off for [domain, db]
2. This change is deployed
3. task from #1 is still running
4. this task gets kicked off again X times for [domain, casetype1, db], casetype2....

In reality I think there's a critical section later that would make this a non-issue, but I wanted to verify if that is safe

Reply:
The task has a built in timer that will cause it to exit after 23 hours so I think the chance of having overlap is limited.

Reply:
```suggestion
    '{domain}-{case_type}-{db}',
```